buku

```diff
@@ -5778,11 +5772,14 @@ def parse_range(tokens: Optional[str | Sequence[str] | Set[str]],  # Optional[st
     for token in ([tokens] if isinstance(tokens, str) else tokens):
         for idx in token.split(','):
             if is_int(idx):
-                result |= {int(idx)}
+                result |= ({int(idx)} if not idx.startswith('-') or maxidx is None else
+                           set(range(maxidx, max(0, maxidx + int(idx)), -1)))
```

Comment:
Without `maxidx` negative numbers are used as-is (and possibly rejected by the validator)